From 16a192d54760fefc7171fd6454f5d27507625f86 Mon Sep 17 00:00:00 2001
From: Chris Wilson <chris@chris-wilson.co.uk>
Date: Wed, 29 Sep 2010 21:19:54 +0100
Patch-mainline: 2.6.37
References: fate#310916
Git-commit: f394940b8d275064f080a59dac636688dae3531a
Subject: [PATCH 1513/2588] drm/i915: Remove redundant deletion of
 obj->gpu_write_list

At that point as the object is no longer in any GPU write domain it must
not be on the list, so the list_del() is redundant.

Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>
(cherry picked from commit f394940b8d275064f080a59dac636688dae3531a)

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 drivers/gpu/drm/i915/i915_gem.c |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_gem.c b/drivers/gpu/drm/i915/i915_gem.c
index 7667fcc..33b708c 100644
--- a/drivers/gpu/drm/i915/i915_gem.c
+++ b/drivers/gpu/drm/i915/i915_gem.c
@@ -3860,8 +3860,6 @@ i915_gem_do_execbuffer(struct drm_device *dev, void *data,
 		if (obj->write_domain)
 			list_move_tail(&obj_priv->gpu_write_list,
 				       &dev_priv->mm.gpu_write_list);
-		else
-			list_del_init(&obj_priv->gpu_write_list);
 
 		trace_i915_gem_object_change_domain(obj,
 						    obj->read_domains,
-- 
1.7.6

